backend/tests/integration/conftest.py

```diff
@@ -156,212 +145,44 @@ def es_clear(es):
-            record = create_record_factory(
-                'lit',
-                with_pid=True,
-                with_index=False,
-            )
-    """
-
-    def _create_record_factory(
-        record_type,
-        data=None,
-        with_pid=True,
-        with_indexing=False,
-        with_validation=False,
-    ):
-        control_number = random.randint(1, 2_147_483_647)
-        if with_validation:
-            data = faker.record(record_type, data)
-        record = RecordMetadataFactory(
-            record_type=record_type, data=data, control_number=control_number
-        )
-
-        if with_pid:
-            record._persistent_identifier = PersistentIdentifierFactory(
-                object_uuid=record.id,
-                pid_type=record_type,
-                pid_value=record.json["control_number"],
-            )
-
-        if with_indexing:
-            index = base_app.config["PID_TYPE_TO_INDEX"][record_type]
-            record._index = es_clear.index(
-                index=get_index_alias(index),
-                id=str(record.id),
-                body=record.json,
-                params={},
-            )
-            current_search.flush_and_refresh(index)
-        return record
-
-    return _create_record_factory
-
-
-@pytest.fixture(scope="function")
-def create_pidstore(db):
-    def _create_pidstore(object_uuid, pid_type, pid_value):
-        return PersistentIdentifierFactory(
-            object_uuid=object_uuid, pid_type=pid_type, pid_value=pid_value
-        )
-
-    return _create_pidstore
-
+_app_ = namedtuple("APP", ["app", "redis", "cli"])
 
-@pytest.fixture(scope="function")
-def api_client(base_app):
-    """Test client for the base application fixture.
-    Scope: function
-    If you need the database and search indexes initialized, simply use the
-    Pytest-Flask fixture ``client`` instead. This fixture is mainly useful if
-    you need a test client without needing to initialize both the database and
-    search indexes.
-    """
-    with base_app.test_client() as client:
-        yield client
-
-
-@pytest.fixture(scope="function")
-def create_user_and_token(base_app, db, es):
-    """Fixtures to create user and authentication token for given user.
 
-    Examples:
-
-        def test_needs_authentication(base_app, create_user_and_token)
-            user = create_user_and_token()
-    """
-
-    def _create_user_and_token():
-        return AccessTokenFactory()
-
-    return _create_user_and_token
+def _setup_cli(app):
+    """Click CLI runner inside the Flask application."""
+    runner = CliRunner()
+    obj = ScriptInfo(create_app=lambda info: current_app)
+    runner._invoke = runner.invoke
+    runner.invoke = partial(runner._invoke, cli, obj=obj)
+    return runner
 
 
 @pytest.fixture(scope="function")
-def create_user(base_app, db, es):
-    """Fixture to create user.
-
-    Examples:
+def app_clean(base_app, db, es_clear, vcr_config):
```

Comment:
There is a flask fixture which is automatically trying to monkey patch all `app` fixtures which would not work correctly with our setup https://github.com/pytest-dev/pytest-flask/blob/ba8b83f7ce91678704708fdd399264dc57c672ee/pytest_flask/plugin.py#L98

Merging s3 app with other apps will cause 2 big problems: First it will enable s3 on all tests, which will break many of them as they were created long before s3 was introduced in Inspire Fixing this will take ages. Second S3 initialization takes around 0.3s-0.5s so multiply it by 800 tests and we have tests slower by 4-7 minutes